Content.Client/ContextMenu/UI/EntityMenuPresenter.cs

```diff
@@ -0,0 +1,421 @@
+            var orderedStates = entitySpriteStates.ToList();
+            orderedStates.Sort((x, y) => string.CompareOrdinal(x.First().Prototype?.Name, y.First().Prototype?.Name));
+            Elements.Clear();
+            AddToUI(orderedStates);
+
+            var box = UIBox2.FromDimensions(_userInterfaceManager.MousePositionScaled.Position, (1, 1));
+            RootMenu.Open(box);
+        }
+
+        public override void OnKeyBindDown(ContextMenuElement element, GUIBoundKeyEventArgs args)
+        {
+            base.OnKeyBindDown(element, args);
+            if (element is not EntityMenuElement entityElement)
+                return;
+
+            // get an entity associated with this element
+            var entity = entityElement.Entity;
+            entity ??= GetFirstEntityOrNull(element.SubMenu);
+            if (entity == null)
+                return;
+
+            // open verb menu?
+            if (args.Function == ContentKeyFunctions.OpenContextMenu)
+            {
+                _verbSystem.VerbMenu.OpenVerbMenu(entity);
+                args.Handle();
+                return;
+            }
+
+            // do examination?
+            if (args.Function == ContentKeyFunctions.ExamineEntity)
+            {
+                _systemManager.GetEntitySystem<ExamineSystem>().DoExamine(entity);
+                args.Handle();
+                return;
+            }
+
+            // do some other server-side interaction?
+            if (args.Function == EngineKeyFunctions.Use || args.Function == ContentKeyFunctions.AltActivateItemInWorld || args.Function == ContentKeyFunctions.Point ||
+                args.Function == ContentKeyFunctions.TryPullObject || args.Function == ContentKeyFunctions.MovePulledObject)
+            {
+                var inputSys = _systemManager.GetEntitySystem<InputSystem>();
+
+                var func = args.Function;
+                var funcId = _inputManager.NetworkBindMap.KeyFunctionID(func);
+
+                var message = new FullInputCmdMessage(_gameTiming.CurTick, _gameTiming.TickFraction, funcId,
+                    BoundKeyState.Down, entity.Transform.Coordinates, args.PointerLocation, entity.Uid);
+
+                var session = _playerManager.LocalPlayer?.Session;
+                if (session != null)
+                {
+                    inputSys.HandleInputCommand(session, func, message);
+                }
+
+                _verbSystem.CloseAllMenus();
+                args.Handle();
+                return;
+            }
+
+            if (_itemSlotManager.OnButtonPressed(args, entity))
+            {
+                _verbSystem.CloseAllMenus();
+            }
+        }
+
+        private bool HandleOpenContextMenu(in PointerInputCmdHandler.PointerInputCmdArgs args)
+        {
+            if (args.State != BoundKeyState.Down)
+                return false;
+
+            if (_stateManager.CurrentState is not GameScreenBase)
+                return false;
```

Comment:
this seems very ugly to have in this function but im assuming it was like this before so

Reply:
Yeah it was there before and I just kept it. But I'm not sure it's actually needed? 

The `OpenContextMenu` keybinding does not call the bound function while in the lobby. I have no idea whether there would ever be a situation where you need that check to happen. Should I just remove it?